linkcheck/better_exchook2.py

```diff
@@ -40,201 +40,283 @@
+            limit = sys.tracebacklimit
+        n = 0
+        _tb = tb
+
+        def _resolveIdentifier(namespace, id):
+            obj = namespace[id[0]]
+            for part in id[1:]:
+                obj = getattr(obj, part)
+            return obj
+
+        def _trySet(old, prefix, func):
+            if old is not None:
+                return old
+            try:
+                return prefix + func()
+            except KeyError:
+                return old
+            except Exception as e:
+                return prefix + "!" + e.__class__.__name__ + ": " + str(e)
+
+        while _tb is not None and (limit is None or n < limit):
+            f = _tb.tb_frame
+            allLocals.update(f.f_locals)
+            allGlobals.update(f.f_globals)
+            lineno = _tb.tb_lineno
+            co = f.f_code
+            filename = co.co_filename
+            name = co.co_name
+            output('  File "%s", line %d, in %s' % (filename, lineno, name), out=out)
+            if not os.path.isfile(filename):
+                altfn = fallback_findfile(filename)
+                if altfn:
+                    output(
+                        "    -- couldn't find file, trying this instead: " + altfn,
+                        out=out,
+                    )
+                    filename = altfn
+            linecache.checkcache(filename)
+            line = linecache.getline(filename, lineno, f.f_globals)
+            if line:
+                line = line.strip()
+                output('    line: ' + line, out=out)
+                output('    locals:', out=out)
+                alreadyPrintedLocals = set()
+                for tokenstr in grep_full_py_identifiers(parse_py_statement(line)):
+                    splittedtoken = tuple(tokenstr.split("."))
+                    for token in map(
+                        lambda i: splittedtoken[0:i], range(1, len(splittedtoken) + 1)
+                    ):
+                        if token in alreadyPrintedLocals:
+                            continue
+                        tokenvalue = None
+                        tokenvalue = _trySet(
+                            tokenvalue,
+                            "<local> ",
+                            lambda: pretty_print(_resolveIdentifier(f.f_locals, token)),
+                        )
+                        tokenvalue = _trySet(
+                            tokenvalue,
+                            "<global> ",
+                            lambda: pretty_print(
+                                _resolveIdentifier(f.f_globals, token)
+                            ),
+                        )
+                        tokenvalue = _trySet(
+                            tokenvalue,
+                            "<builtin> ",
+                            lambda: pretty_print(
+                                _resolveIdentifier(f.f_builtins, token)
+                            ),
+                        )
```

Comment:
uh, okay wow. i did know we had that complex of an exception handler. if we want to fix formatting in this, maybe we should refactor the actual code a little rather than just show this under the rug. it seems what we're trying to do here is to fix long lines, but there's other issues going on here...

at least i'm not sure this specific hunk improves readability at all...

Reply:
In the context of this PR we can just skip long lines if that is better?

Reply:
maybe, yeah, for now.